### DISCUSSION TITLE
Write EXR Metadata without rewriting entire file

### DISCUSSION THREAD

**Discussion by brycegbrazen** [OpenImageIO/oiio#3412] [Q&A] [answered]
Hi all,

Using the OIIO API, is there any way to read in an EXR file, edit the metadata, and write out only the minimal header change necessary?

I've tried using an ImageBuf object with specmod(), but this seems to require you to write out the whole file again with the .write() function. This can be very slow when iterating over several EXR files and adding metadata to each of them.

Thanks,
Bryce

**Comment by lgritz** [comment, marked as answer]
No, there's not a way to do that. It's not something that is supported by the underlying libOpenEXR. Or, as far as I know, by any of the other image format libraries we depend on, so OIIO never made an API consideration for it.

I think it would be a great feature of an image format -- the ability to edit in-place (or at least add) metadata without touching the rest of the file. But as far as I can tell, none of the file formats or their reader libraries were designed with this in mind.
